lib/npo.src.js

```diff
@@ -77,6 +124,15 @@
 			cycle = timer(scheduling_queue.drain);
 		}
 	}
+	
+	function hasCatch(chain) {
+		for (var i = 0; i < chain.length; i++) {
+			if (chain[i].failure) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
I recommend you run this against the NodeJS tests or the Q ones for the events sicne they check this sort of thing.